### PR TITLE
feat: DeleteAccountUseCase and split UserRepository 

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors
 
- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

we need a use case to request a delete account code

### Solutions

implement it and take some functions from the user repository to the AccountRepository

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by saleniuk — APPROVED** [review]

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt [hunk 1]
```diff
@@ -412,87 +366,6 @@ class UserRepositoryTest {
...
-            shouldFail()
-            verify(arrangement.selfApi)
-                .suspendFunction(arrangement.selfApi::updateEmailAddress)
-                .with(eq("newEmail"))
-                .wasInvoked(exactly = once)
-        }
-    }
-
-    @Test
-    fun givenThereAreUsersWithoutMetadata_whenSyncingUsers_thenShouldUpdateThem() = runTest {
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt:450]: Isn't this test valid anymore? I see it was removed and doesn't seem like it was moved to `AccountRepositoryTest`.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt [hunk 2]
```diff
@@ -412,87 +366,6 @@ class UserRepositoryTest {
...
-        verify(arrangement.userDAO)
-            .suspendFunction(arrangement.userDAO::upsertUsers)
-            .with(matching {
-                it.first().name != null
-            })
-            .wasInvoked(exactly = once)
-    }
-
-    @Test
-    fun givenThereAreNOUsersWithoutMetadata_whenSyncingUsers_thenShouldNOTUpdateThem() = runTest {
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt:475]: Same here. The test is gone 💨

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Looks good.

Just noticed that two tests vanished from `UserRepositoryTest`.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt (hunk 1 above)
Comment (MohamadJaara) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt:450]: i think it was deleted when solving merge conflict with dev, i will add it

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**